### PR TITLE
Collect test run data in Alfred

### PR DESCRIPTION
**Pull request by giorgiosironi**
`green` appears [not to support this log](https://github.com/CleanCut/green/issues/104) but it's useful to get feedback on which tests are taking the longest, or failing more often.

### CODE REVIEW

**Comment by giorgiosironi** [comment]
To solve:
```
17:15:21 Traceback (most recent call last):
17:15:21   File "/ext/jenkins/workspace/er_PR-273-FTO5INUDHBOUZ6O75B56@2/.tox/py27/bin/pytest", line 7, in <module>
17:15:21     from pytest import main
17:15:21   File "/ext/jenkins/workspace/er_PR-273-FTO5INUDHBOUZ6O75B56@2/.tox/py27/local/lib/python2.7/site-packages/pytest.py", line 28, in <module>
17:15:21     from _pytest.python_api import approx, raises
17:15:21   File "/ext/jenkins/workspace/er_PR-273-FTO5INUDHBOUZ6O75B56@2/.tox/py27/local/lib/python2.7/site-packages/_pytest/python_api.py", line 6, in <module>
17:15:21     from more_itertools.more import always_iterable
17:15:21 ImportError: cannot import name always_iterable
17:15:21 ERROR: InvocationError: '/ext/jenkins/workspace/er_PR-273-FTO5INUDHBOUZ6O75B56@2/.run-tests-ci.sh py27'
```

**Comment by giorgiosironi** [comment]
No idea why there are multiple processes involved, I would have expected everything to run into one by default as these tests are not designed for concurrency. Switching back to https://github.com/elifesciences/builder/pull/273 to see if that helps.

**Comment by giorgiosironi** [comment]
`congratulations :)` :tada:

**Comment by giorgiosironi** [comment]
Sample result: https://alfred.elifesciences.org/job/pull-requests-projects/job/builder/job/PR-273/9/testReport/
(all tests are shown twice, because they run twice, on 2.7 and 3.5)

**Comment by giorgiosironi** [comment]
Pipeline looks like this

but I'm going to move also `Static checking` in parallel with the rest.

**Comment by giorgiosironi** [comment]
Infinite scaling on the cloud -.-
```
  <Error>
    <Type>Sender</Type>
    <Code>Throttling</Code>

    <Message>Rate exceeded</Message>
  </Error>
```

**Comment by giorgiosironi** [comment]
At the moment favors stability over speed. I have plans on exponential backoff and other improvements to be able to run more things in parallel.

File: requirements.txt [hunk 1]
```diff
@@ -12,6 +12,7 @@ more-itertools==4.1.0
 netaddr==0.7.19
 pyflakes==1.6.0
 pylint==1.8.2
+pytest==3.5.0
```
Comment (seanwiseman) [inline requirements.txt]: 🍾

File: .run-tests-ci.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+if [ -z "$envname" ]; then
+    echo "Must pass a label for test artifact file e.g. py27"
+    exit 1
+fi
+
+export PYTHONPATH="src"
+coverage run -m pytest \
+    -s \
+    --junitxml=build/pytest-$envname.xml \
+    src/tests src/integration_tests
```
Comment (seanwiseman) [inline .run-tests-ci.sh]: Can alternatively be defined in a `pytest.ini` file in the project root with:
```
[pytest]
testpaths = src/test src/integration_tests
```

Comment (giorgiosironi) [reply .run-tests-ci.sh]: Ok, may move it there later, although when using it locally you may want to run just `src/tests` due to the heaviness of the `integration_tests`.

**Pull request merged**